CPS-XXXX/README.md

```diff
@@ -0,0 +1,87 @@
+---
+CPS: ?
+Title: Universal JSON Encoding for Domain Types
+Status: Open
+Category: Tools
+Authors:
+    - Vladimir Kalnitsky <vladimir@mlabs.city>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/742
+Created: 2024-01-10
+---
+
+<!-- Existing categories:
+
+- Meta     | For meta-CIPs which typically serves another category or group of categories.
+- Wallets  | For standardisation across wallets (hardware, full-node or light).
+- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+- Metadata | For proposals around metadata (on-chain or off-chain).
+- Tools    | A broad category for ecosystem tools not falling into any other category.
+- Plutus   | Changes or additions to Plutus
+- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+-->
+
+## Abstract
+<!-- A short (\~200 word) description of the target goals and the technical obstacles to those goals. -->
+
+dApps require the use of communication protocols that facilitate data transfer between:
+
+- query layer providers and dApps
+- wallets and dApps
+- wallets and query layer providers
+- dApp frontends and backends
+- different dApps
+- Cardano blockchain indexers and their API consumers
+
+Usually these protocols include a specification of Cardano domain types, i.e. ledger block and all of its sub-components.
+
+## Problem
+
+Cardano domain types have canonical CDDL definitions (for every era), but when it comes to use in web apps, where JSON is the universally accepted format, there is no definite standard.
```

Comment:
This document should probably reference:
1. The discussion of converting between JSON and CBOR in the CBOR spec: https://www.rfc-editor.org/rfc/rfc8949.html#name-converting-data-between-cbo
2. The discussion of using CDDL to specify JSON in the CDDL spec: https://datatracker.ietf.org/doc/html/rfc8610#appendix-E